File: signature/styled_components.go
package signature

var StyledComponentsSignature = Signature{
	Name:        "styled components",
	Description: "styled-components is the result of wondering how we could enhance CSS for styling React component systems.",
	Cpe:         "",

	DetectPattern: DetectPattern{
		URLs: []string{
			"styled-components",
		},
		Bodies: []string{
			"data-styled=\"true\"",
			"data-styled-version",
		},
	},
	VersionPattern: VersionPattern{
		Bodies: []string{
			`data-styled-version="(\d+\.\d+\.\d+)"`,
			`t\["data-styled-version"\]="(\d+\.\d+\.\d+)"`,
			`\[data-styled-version="(\d+\.\d+\.\d+)"\]`,
			`setAttribute\("data-styled-version","(\d+\.\d+\.\d+)"\)`,
			`<style data-styled="true" data-styled-version="(\d+\.\d+\.\d+)">`,
		},
	},
}

File: src/signatures/technologies/styled_components.ts
import type { Signature } from "../_types.js";
import { reactSignature } from "./react.js";

export const styledComponentsSignature: Signature = {
  name: "styled-components",
  description:
    "Styled components is a CSS-in-JS styling framework that uses tagged template literals in JavaScript.",
  rule: {
    confidence: "high",
    bodies: [
      "(?<![\\w-])data-styled(?![\\w-])",
      "(?<![\\w-])sc-component-id(?![\\w-])",
    ],
    javascriptVariables: {
      styled: "",
    },
  },
  impliedSoftwares: [reactSignature.name],
};
